custom-css-analysis-and-migration/migrateCssToAppearance.ts

```diff
@@ -201,32 +255,17 @@ function sortDeclarations(rule: Rule) {
     }
   });
 }
-function sortClassesInSelector(rule: any): void {
-  // sort the classes in the first selector
-  const classes = rule.selectors[0].trim().split(".").filter(Boolean).sort();
-  const sortedSelector = "." + classes.join(".");
-  rule.selectors[0] = sortedSelector;
-}
 
-function sortRules(rules: Rule[]): Rule[] {
-  return rules.sort((a: Rule, b: Rule) => {
-    if (a.type !== "rule" || b.type !== "rule") return 0;
-
-    // sort rules by first selector
-    const aSelector = a.selectors ? a.selectors[0] : undefined;
-    const bSelector = b.selectors ? b.selectors[0] : undefined;
-    if (
-      aSelector === bSelector ||
-      aSelector === undefined ||
-      bSelector === undefined
-    ) {
-      return 0;
-    } else if (aSelector > bSelector) {
-      return 1;
-    } else {
-      return -1;
-    }
-  });
+function sortClassesInSelector(rule: any): void {
+  // sort the classes in the first selector if appropriate
+  if (!rule.selectors || !rule.selectors.length) return;
+  var selector = rule.selectors[0].trim();
+  if (/^.[A-Za-z][-.A-Za-z0-9]*$/.test(selector))
```

Comment:
I'm puzzled.
1. Why do we want to sort selectors at all?
2. If I'm reading the calling code correctly, rule is a CssRule object (could we declare that?). These are not expected to have a selectors property, so maybe this code will never do anything?
3. Since I can't find any doc on this property, I'm not sure what it contains. For example, given an overall selector like ".bloom-page.A5Portrait .marginBox", is rule.selectors[0].trim() supposed to return ".bloom-page.A5Portrait"? So this would convert it to ".A5Portrait.bloom-page .marginBox"?
4. Is the first dot actually meant to match a dot, rather than any character? (Should it have a backslash?) So for example it then wouldn't attempt to sort ":not(.frontCover).bloom-page.front-matter"? (Of course it won't anyway because the parens don't match.)
5. If we need the trim(), it's presumably because rule.selectors[0] includes the space separating it from the next component of the selector. If that's true, so we need to put that space back below?
6. Assuming there IS a reason to sort them, will it matter than a selector like .bloom-page.A5Portrait:not(.front-matter):not(.back-matter) does not get changed (because the parens don't match)? (If you need to make it handle that, remember that splitting at dots and sorting will do something horrible to :not(.front-matter))

Reply:
1. I don't know that we need to, but JohnH's original code was doing this, albeit not as safely.
2. I haven't been able to find out anything about the classes used here.  "bun" (which is how JohnH set things up) has all sorts of things built in that don't have any intellesense type operation.
3. The selector property is an array of strings that contain the comma separated values from the CSS selector of the rule.  The trim() operation would only remove spaces around a comma or the overall beginning or end of the CSS selector.
4. good catch.  :-)  Yes, I only try to sort extremely simple selectors:  no spaces, no colons, no parentheses...
5. We don't need to put the space back.  The stringify method imported form css has its own ideas of formatting.
6. I think the only reason to sort the selector is to make simple cases easier to compare visually by human being.  This may not be a compelling reason.